### DISCUSSION TITLE
High CPU usage

### DISCUSSION THREAD

**Discussion by nunog-costa** [bluenviron/mediamtx#4055] [Questions]
### Question

Hi,

Currently running Media MTX in a Azure Windows VM (Standard D2s v3 - 2 vcpus, 8 GiB memory) and I have several machines with their own RTSP servers pushing streams to MediaMTX via ffmpeg. I'm relying on MediaMTX to convert them to WebRTC so they can be watched in a browser.

I'm doing my initial tests and noticed that when pushing just one stream and watching it from 8 different computers/browsers the server's CPU is already at around 50%. I know my server specs are quite weak but is there anything I can do to optimize the performance? Each user watching the same stream seems to be linearly increasing the CPU consumption.

My customer wants to livestream around 100 cameras, is MediaMTX able to handle this?

Thanks in advance,
Nuno

**Comment by adzmailru** [comment]
with "on demand" or not? with or without ffmpeg?

**Reply by nunog-costa** [reply]
Apologies for the late reply.

sourceOnDemand is set to no and I have nothing on runOnDemand.
 
My workflow is as follows:
 
- Remote into azure server, start MediaMTX
- Start local RTSP server in my local computer
- Push RTSP stream to MediaMTX using ffmpeg
 
The ffmpeg command I'm using is the following, but I've been trying different parameters and measuring their impact:
 
_ffmpeg -rtsp_transport tcp -i <RTSP_STREAM_URL> -c:v libx264 -profile:v baseline -level 3.1 -preset veryfast -tune zerolatency -b:v 2M -maxrate 2M -bufsize 4M -g 60 -c:a libopus -b:a 64k -ar 48000 -ac 2 -reorder_queue_size 256 -vsync 1 -async 1 -f rtsp <MEDIAMTX_STREAM_URL>_

**Comment by aler9** [comment]
Hello, what is causing load on the CPU is not MediaMTX but the FFmpeg process that converts stream into H264. The number of readers and browsers is not relevant and doesn't influence the load.

Converting videos is a resource-hungry process. Most cloud services provide some kind of utility or optimized instances to handle this process in a better way. In case of Azure there's the [NV family](https://learn.microsoft.com/en-us/azure/virtual-machines/sizes/gpu-accelerated/nv-family).
